examples/basic/src/main/java/com/soundcloud/lightcycle/sample/basic/BaseActivity.java

```diff
@@ -0,0 +1,12 @@
+package com.soundcloud.lightcycle.sample.basic;
+
+import android.support.v7.app.AppCompatActivity;
+
+import com.soundcloud.lightcycle.ActivityLightCycle;
+import com.soundcloud.lightcycle.LightCycleBaseClass;
+import com.soundcloud.lightcycle.LightCycleDispatcher;
+
+@LightCycleBaseClass
+public abstract class BaseActivity extends AppCompatActivity
+        implements LightCycleDispatcher<ActivityLightCycle<AppCompatActivity>> {
```

Comment:
Yes. I made sure that it works with things like generics too (as you can see with that test)